Src/Newtonsoft.Json.Schema/JSchema.cs

```diff
@@ -114,6 +113,16 @@ internal IEnumerable<PatternSchema> GetPatternSchemas()
         public bool? Valid { get; set; }
 
         /// <summary>
+        /// Gets or sets the $ref
+        /// </summary>
+        public Uri Reference { get; set; }
+
+        /// <summary>
+        /// Gets the flag indicating this schema is resolved.
+        /// </summary>
+        public bool IsReferenceResolved { get; internal set; }
```

Comment:
What happens if you try to validate JSON using a schema that has a reference that is not resolved? I think an error being thrown would be a good idea.

Reply:
Should this be public? It is kind of redundant information because a schema with a resolved reference will always have a null `Reference` property.

Reply:
Ah, I've changed the behaviour when a reference is resolved to set `Reference` back to the original reference `$ref`. So exposing another property to help APIs know if the schema was resolved or not. 

Throwing an error when a reference fails to resolve may make the API too noisy and difficult to nurse when dealing with schema sets that are out of your control.

Reply:
An error should be thrown if you try to validate with a schema that is just a reference. It doesn't make sense at validation time.

Reply:
Oh, is this library exclusively used for validating schemas?

Reply:
No, you can use it to generate schemas from .NET types, or build a schema manually using JSchema and then write it as JSON. Manually setting `Reference` seems like it would make sense in those situations but once you need to validate JSON then it doesn't.